presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -83,6 +102,31 @@ protected Void visitJoin(Join node, MaterializedViewPlanValidatorContext context
 
                 break;
 
+            case LEFT:
+                if (!node.getCriteria().isPresent()) {
+                    throw new SemanticException(NOT_SUPPORTED, node, "Outer join with no criteria is not supported for materialized view.");
+                }
+
+                joinCriteria = node.getCriteria().get();
+                if (!(joinCriteria instanceof JoinOn)) {
+                    throw new SemanticException(NOT_SUPPORTED, node, "Only join-on is supported for materialized view.");
+                }
+
+                process(node.getLeft(), context);
+
+                boolean wasWithinOuterJoin = context.isWithinOuterJoin();
+                context.setWithinOuterJoin(true);
```

Comment:
`wasWithinOuterJoin` is only true for the first time. Is that an issue? We only allow one outer join in a materialized view?

Reply:
Yeah this is what we want, checking what tables are in some part of outer join. That means those values could become null

Reply:
Let's add a comment to indicate that this is intended.